cubic-server/protocol/ClientPackets.cpp

```diff
@@ -23,6 +23,53 @@ std::shared_ptr<std::vector<uint8_t>> protocol::createStatusResponse(const Statu
     return packet;
 }
 
+std::shared_ptr<std::vector<uint8_t>> protocol::createLoginSuccess(const LoginSuccess &in)
+{
+    std::vector<uint8_t> payload;
+    serialize(payload, in.uuid, addUUID,
+        in.username, addString,
+        in.numberOfProperties, addVarInt,
+        in.name, addString,
+        in.value, addString,
+        in.isSigned, addBoolean);
+    if (in.isSigned)
+        serialize(payload, in.signature.value(), addString);
+    auto packet = std::make_shared<std::vector<uint8_t>>();
+    finalize(*packet.get(), payload, (int32_t)ClientPacketID::LoginSuccess);
+    return packet;
+}
+
+std::shared_ptr<std::vector<uint8_t>> protocol::createLoginPlay(const LoginPlay &in)
+{
+    std::vector<uint8_t> payload;
+    serialize(payload,
+        in.entityID, addVarInt,
+        in.isHardcore, addBoolean,
+        in.gamemode, addByte,
+        in.previousGamemode, addByte,
+        in.dimensionCount, addVarInt,
+        in.dimensionNames, addArray<std::string, addString>,
+        in.registryCodec, addNBT<nbt::Compound>,
```

Comment:
Wait, you made it a template ? why ?
it should already call the corresponding `serialize` method of the derived class.

Reply:
With @huntears we decided to do it like this because it wasn't working at all, but if you can do it without it would be great !

Reply:
Ok, because of how `protocol::serialize` is made, it's not possible.
Well, technicaly it is, but this will let so many error pass that it's not a good solution

Reply:
Problem is that I cannot pass a `nbt::Base` like this to the function, the compiler will complain, making it a template solves this problem altogether, and it should just work since templates are also used by `addArray`.